packages/flutter/lib/src/material/dialog.dart

```diff
@@ -543,63 +543,6 @@ class SimpleDialog extends StatelessWidget {
   }
 }
 
-class _DialogRoute<T> extends PopupRoute<T> {
-  _DialogRoute({
-    @required this.theme,
-    bool barrierDismissible = true,
-    this.barrierLabel,
-    @required this.child,
-    RouteSettings settings,
-  }) : assert(barrierDismissible != null),
-       _barrierDismissible = barrierDismissible,
-       super(settings: settings);
-
-  final Widget child;
-  final ThemeData theme;
-
-  @override
-  Duration get transitionDuration => const Duration(milliseconds: 150);
-
-  @override
-  bool get barrierDismissible => _barrierDismissible;
-  final bool _barrierDismissible;
-
-  @override
-  Color get barrierColor => Colors.black54;
-
-  @override
-  final String barrierLabel;
-
-  @override
-  Widget buildPage(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
-    return new SafeArea(
-      child: new Builder(
-        builder: (BuildContext context) {
-          final Widget annotatedChild = new Semantics(
-            child: child,
-            scopesRoute: true,
-            explicitChildNodes: true,
-          );
-          return theme != null
-            ? new Theme(data: theme, child: annotatedChild)
-            : annotatedChild;
-        }
-      ),
-    );
-  }
-
-  @override
-  Widget buildTransitions(BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
-    return new FadeTransition(
-      opacity: new CurvedAnimation(
-        parent: animation,
-        curve: Curves.easeOut
-      ),
-      child: child
-    );
-  }
-}
-
 /// Displays a dialog above the current contents of the app.
 ///
```

Comment:
Also describe why this function specifically exists (e.g. that it's dismissable by tapping outside the dialog etc).